org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/JvmTypeReferenceUtil.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2024 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.common.types.util;
+
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.xtext.common.types.JvmTypeReference;
+
+/**
+ * @author Lorenzo Bettini - Initial contribution and API
+ */
+public class JvmTypeReferenceUtil {
+
+	private static class ExpectedClassMarker extends AdapterImpl {
+		@Override
+		public boolean isAdapterForType(Object type) {
+			return type == ExpectedClassMarker.class;
+		}
+	}
+
+	private static class ExpectedInterfaceMarker extends AdapterImpl {
```

Comment:
AdapterImpl has a field `Notifier target` that is maintained. I don't think we need that and it should be fine use an implementation similar to org.eclipse.xtext.xbase.validation.ReadAndWriteTracking.READMARKER

Reply:
OK, so simply check the singleton is there.
I'll do that

Reply:
I now use the same strategy of `org.eclipse.xtext.xbase.validation.ReadAndWriteTracking.READMARKER` in https://github.com/eclipse/xtext/pull/2924/commits/4b407b41bfab5d77935627d95dbb383aefdba417